src/MouseAdapter.js

```diff
@@ -309,20 +352,231 @@ define(function(require) {
+         * @param {Array.<number>} array The array of data to analyze for this check
+         * @return {boolean} Returns true if conditions were satisfied to indicate the
+         * user performed a scroll event, else false
+         * @private
+         */
+        _detectNegativeSpike: function(array) {
+            var lastPoint = this._getDeltaArrayIndex(-1);
+            // Check if the current event delta is low.  Nearly all of the negative spikes 
+            // seem to reach between 0.5-2.5, so make sure the point is low enough
+            if (array[lastPoint] <= 5) {
+                var recentPoint2 = this._getDeltaArrayIndex(-2);
+                var recentPoint3 = this._getDeltaArrayIndex(-3);
+                // Check to see if the previous two points are significantly higher.
+                // Checks last TWO points instead of just last one because the deltas
+                // seem to be prone to intermittent noisy spikes which can create false
+                // positives.
+                var isFirstPointHigherMagnitude = array[recentPoint2] / array[lastPoint] > 2;
+                var isSecondPointHigherMagnitude = array[recentPoint3] / array[lastPoint] > 2;
+                var isFirstPointHigherValue = array[recentPoint2] - array[lastPoint] > 2;
+                var isSecondPointHigherValue = array[recentPoint3] - array[lastPoint] > 2;
+                if ((isFirstPointHigherMagnitude && isFirstPointHigherValue) &&
+                    (isSecondPointHigherMagnitude && isSecondPointHigherValue)) {
+                    return true;
+                }
+            }
+            return false;
+        },
+
+        /**
+         * A helper function for calculating the index of the desired data given its 
+         * relative index.  Negative numbers can be used to access the N'th most recent
+         * data point.
+         *
+         * For example: 
+         *   getDeltaArrayIndex(0) will return the index of the oldest data point
+         *   getDeltaArrayIndex(DELTA_ARRAY_SIZE-1) will return the most recent data index.
+         *   getDeltaArrayIndex(-1) will return the most recent data point index.
+         *   getDeltaArrayIndex(DELTA_ARRAY_SIZE) will return the oldest data point index.
+         * @type {Function}
+         * @return {number} The properly offset index of the requested data point.
+         */
+        _getDeltaArrayIndex: function(index) {
+            var result = (this._deltaArrayIndex+index);
+            if (index < 0) {
+                if (index < -this.DELTA_ARRAY_SIZE) {
+                    throw new Error('Array index out of bounds!');
+                }
+                result += this.DELTA_ARRAY_SIZE;
+            }
+            else {
+                if (index >= this.DELTA_ARRAY_SIZE) {
+                    throw new Error('Array index out of bounds!');
+                }
+            }
+            result = result % this.DELTA_ARRAY_SIZE;
+            return result;
```

Comment:
Would be good to see at least one test just to ensure this doesn't have an off-by-one error (at each end).